tests/test-groups.py

```diff
@@ -300,6 +300,22 @@ def test_reduce_prod():
     assert f1.to_list() == [["blue", "green", "red"],
                             [2, -7, 65]]
 
+def test_reduce_prod_same_column():
+    # See issue #3390
+    f0 = dt.Frame({"ints" : [0, 1, 0, 0, 1, 2]})
+    f1 = f0[:, {"prod" : prod(f.ints)}, f.ints]
+    frame_integrity_check(f1)
+    assert_equals(f1, dt.Frame({"ints" : [0, 1, 2], "prod" : [0, 1, 2]/dt.int64}))
+
+
+def test_reduce_prod_same_column1():
+    # See issue #3390
+    f0 = dt.Frame({"ints" : [0, 1, 2, 2, 1, 2]})
+    f1 = f0[:, {"prod" : prod(f.ints)}, f.ints]
+    frame_integrity_check(f1)
+    assert_equals(f1, dt.Frame({"ints" : [0, 1, 2], "prod" : [0, 1, 8]/dt.int64}))
```

Comment:
I guess this test is enough, as the one above only tests a very simple case when `prod(ints) == ints`.